File: Compose/YABA/app/src/main/java/dev/subfly/yaba/core/state/detail/canvmark/CanvmarkDetailEvent.kt
package dev.subfly.yaba.core.state.detail.canvmark

import dev.subfly.yaba.core.webview.CanvasHostMetrics
import dev.subfly.yaba.core.webview.CanvasHostStyleState
import dev.subfly.yaba.core.webview.WebShellLoadResult

sealed interface CanvmarkDetailEvent {
    data class OnInit(val bookmarkId: String) : CanvmarkDetailEvent
    data class OnSave(val sceneJson: String) : CanvmarkDetailEvent
    data class OnWebInitialContentLoad(val result: WebShellLoadResult) : CanvmarkDetailEvent
    data class OnCanvasMetricsChanged(val metrics: CanvasHostMetrics) : CanvmarkDetailEvent
    data class OnCanvasStyleStateChanged(val style: CanvasHostStyleState) : CanvmarkDetailEvent
    data object OnToggleCanvasOptionsSheet : CanvmarkDetailEvent
    data object OnDismissCanvasOptionsSheet : CanvmarkDetailEvent
    data object OnPickImageFromGallery : CanvmarkDetailEvent
    data object OnCaptureImageFromCamera : CanvmarkDetailEvent
    data object OnConsumedPendingImageInsert : CanvmarkDetailEvent
    data object OnDeleteBookmark : CanvmarkDetailEvent
    data class OnScheduleReminder(
        val title: String,
        val message: String,
        val triggerAtEpochMillis: Long,
    ) : CanvmarkDetailEvent
    data object OnCancelReminder : CanvmarkDetailEvent
    data class OnExportImageReady(val bytes: ByteArray, val extension: String) : CanvmarkDetailEvent
}

File: Darwin/YABACore/State/Detail/Canvmark/CanvmarkDetailEvent.swift
//
//  CanvmarkDetailEvent.swift
//  YABACore
//

import Foundation

public enum CanvmarkDetailEvent: Sendable {
    case onInit(bookmarkId: String)
    case onSave(sceneJson: String)
    case onWebInitialContentLoad(resultJson: String?)
    case onCanvasMetricsChanged(metricsJson: String?)
    case onCanvasStyleStateChanged(styleJson: String?)
    case onToggleCanvasOptionsSheet
    case onDismissCanvasOptionsSheet
    case onPickImageFromGallery
    case onCaptureImageFromCamera
    case onConsumedPendingImageInsert
    case onDeleteBookmark(bookmarkId: String)
    case onScheduleReminder(titleKey: String, messageKey: String, fireAt: Date)
    case onCancelReminder
    case onExportImageReady(Data, fileExtension: String)
    case saveScene(bookmarkId: String, sceneData: Data)
}
